src/modules/share/share.controller.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  Controller,
+  HttpCode,
+  Get,
+  Put,
+  Param,
+  Query,
+  Post,
+  Body,
+  Res,
+  HttpStatus,
+  NotFoundException,
+} from '@nestjs/common';
+import { ApiOperation, ApiTags, ApiOkResponse } from '@nestjs/swagger';
+import { ShareUseCases } from './share.usecase';
+import { User as UserDecorator } from '../auth/decorators/user.decorator';
+import { CreateShareDto } from './dto/create-share.dto';
+import { Response } from 'express';
+import { GetDownFilesDto } from './dto/get-down-files.dto';
+import { FileUseCases } from '../file/file.usecase';
+import { FolderUseCases } from '../folder/folder.usecase';
+import { UserUseCases } from '../user/user.usecase';
+import { Public } from '../auth/decorators/public.decorator';
+import { UpdateShareDto } from './dto/update-share.dto';
+import { User } from '../user/user.domain';
+
+@ApiTags('Share')
+@Controller('storage/share')
+export class ShareController {
+  constructor(
+    private shareUseCases: ShareUseCases,
+    private fileUseCases: FileUseCases,
+    private folderUseCases: FolderUseCases,
+    private userUseCases: UserUseCases,
+  ) {}
+
+  @Get('/list')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Get share list',
+  })
+  @ApiOkResponse({ description: 'Get all shares in a list' })
+  async listShares(
+    @UserDecorator() user: User,
+    @Query('page') page: string,
+    @Query('perPage') perPage: string,
+  ) {
+    const shares = await this.shareUseCases.listByUserPaginated(
+      user,
+      parseInt(page) || 1,
+      parseInt(perPage) || 50,
+    );
+    return shares;
+  }
+
+  @Get('/:token')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Get share by token',
+  })
+  @ApiOkResponse({ description: 'Get share' })
+  @Public()
+  async getShareByToken(
+    @UserDecorator() user: User,
+    @Param('token') token: string,
+  ) {
+    if (user && !user.id && user.email) {
```

Comment:
This is a bit confusing and seems like business logic, what is doing this check?